src/MDMScheduler.m

```diff
@@ -81,16 +81,33 @@ - (MDMSchedulerActivityState)activityState {
 }
 
 - (void)addPlan:(NSObject<MDMPlan> *)plan to:(id)target {
+  [self commonAddPlan:plan toTarget:target log:nil];
+}
+
+- (void)addPlan:(NSObject<MDMPlan> *)plan named:(NSString *)name to:(id)target {
+  NSParameterAssert(name.length > 0);
+  [self commonAddPlan:plan toTarget:target log:[[MDMTransactionLog alloc] initWithPlans:@[plan] target:target name:name removal:FALSE]];
```

Comment:
@jverkoey is "transaction log" as a concept still useful? I find it to be a past vestige that we can now remove. You'd still need a data holder to hold the Plan's metadata (name mostly). But it's pretty redundant to have both `addPlan:plan` and `TransactionLog initWithPlan:plan`